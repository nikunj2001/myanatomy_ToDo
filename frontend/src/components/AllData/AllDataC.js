import React, { Component } from 'react'
import { withRouter } from "react-router-dom";
import { connect } from 'react-redux';
import PropTypes from "prop-types";
// import { deleteTask } from '../store/asyncMethods/taskMethods';
import { BsFillPenFill, BsTrash } from "react-icons/bs";
class AllDataC extends Component {
  constructor(props) {
    super(props);
    this.state = {
      task: props.data,
    }
  }
  static getDerivedStateFromProps(props, state) {
    if (props.data.length != state.task.length) {
      return {
        task: props.data
      }
    }
    return null;
  }
  clickEdit = (task) => {
    this.props.history.push({
      pathname: "/editTask",
      state: { task }
    });
  }
  deleteClick = (task) => {
    this.props.deleteTask(task._id);
  }
  render() {

    return (
      <>
        {this.state.task.length > 0 ?
          <table border='1' cellPadding='5' cellSpacing='5' width="75%" >
            <thead key="thead">
              <tr key="trow" >
                <th>Task</th>
                <th>Description </th>
                <th>Status</th>
                <th>Change Status</th>
              </tr>
            </thead>
            <tbody key="tbody">
              {this.state.task.map(t => (
                <tr key={t._id} >
                  <td>{t.task}</td>
                  <td>{t.description}</td>
                  <td>{t.status}</td>
                  <td>
                    <button onClick={() => this.clickEdit(t)} className='btn' id='edit-btn'  ><BsFillPenFill className='btns' />
                    </button>
                    <button className='btn' onClick={() => this.deleteClick(t)} id='delete-btn' ><BsTrash className='btns' /></button></td>
                </tr>
              ))
              }
            </tbody>
          </table> : <><h5>No Task Found</h5></>
        }
      </>
    )
  }
}
AllDataC.propTypes = {
  deleteTask: PropTypes.func.isRequired,
  data: PropTypes.array.isRequired
}

AllDataC.defaultProps = {
  deleteTask: () => { },
  data: []

}
export default AllDataC;
