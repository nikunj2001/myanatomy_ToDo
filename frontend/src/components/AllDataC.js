import React, { Component } from 'react'
import {withRouter} from "react-router-dom";
import { connect } from 'react-redux';
import { deleteTask } from '../store/asyncMethods/taskMethods';
import {BsFillPenFill,BsTrash} from "react-icons/bs";
// import {mapDispatchToProps,mapStateToProps} from "./AllData/props";
import PropTypes from "prop-types";
// const mapStateToProps=(state)=>{
//   const {TaskReducer} = state;
//   const {loading,task,deleteT}=TaskReducer;
//   return{
//       loading,task,deleteT
//   }
// }
const mapDispatchToProps=(dispatch)=>{
  return{
    deleteTask:(id)=>dispatch(deleteTask(id)),
    unset:()=>dispatch({type:"UNSET_TASK"}),
    closeLoader:()=>dispatch({type:"CLOSE_LOADER"}),
    closeDelete:()=>dispatch({type:"UNSET_DELETE"}),
    // getTasks:()=>dispatch(fetchTasks())
  }
}
 class AllDataC extends Component {
  constructor(props){
        super(props);
        this.state={
          task:props.data,
        }
    }
   
    static getDerivedStateFromProps(props,state){
      if(props.data.length!=state.task.length){
        return {
        task:props.data
      }
    }
    return null;
    }
  render() {
    const {task}=this.state;
    const   clickEdit=(task)=>{
      this.props.history.push({
        pathname:"/editTask",
        state:{task}
    });
}
const deleteClick=(task)=>{
  console.log(task);
        this.props.deleteTask(task._id);
}
    return (
      <>
     {this.state.task.length>0?<table border='1' cellPadding='5' cellSpacing='5' width="75%" >
       <thead>
                        <tr>
                           <th>Task</th>
                           <th>Description </th>
                           <th>Status</th>         
                           <th>Change Status</th>         
                        </tr>      
       </thead>
      <tbody>

                    {this.state.task.map(t=>(
                        <tr  key={t._id} >
                            <td>{t.task}</td>
                            <td>{t.description}</td>
                            <td>{t.status}</td>
                            <td>
                              <button onClick={()=>clickEdit(t)} className='btn' id='edit-btn'  ><BsFillPenFill className='btns' />
                              </button>
                              <button className='btn' onClick={deleteClick.bind(this,t)} id='delete-btn' ><BsTrash className='btns' /></button></td>
                        </tr>
                    ))
                    }
      </tbody>

              </table>:<><h5>No Task Found</h5></>
  }
      </>
    )
  }
}

AllDataC.propTypes = {
    deleteTask:PropTypes.func.isRequired,
    unset:PropTypes.func.isRequired,
    closeLoader:PropTypes.func.isRequired,
    closeDelete:PropTypes.func.isRequired,
    loading:PropTypes.bool.isRequired,
    task:PropTypes.string.isRequired,
    deleteT:PropTypes.string.isRequired,
}

AllDataC.defaultProps = {
  deleteTask: () => {},
  unset: () => {},
  closeLoader: () => {},
  closeDelete: () => {}
}
const ReduxApp = connect(null,mapDispatchToProps)(AllDataC)
export default withRouter(ReduxApp);
