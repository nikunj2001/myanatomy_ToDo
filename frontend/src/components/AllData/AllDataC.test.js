import React from 'react';
import { shallow, configure } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"
import AllDataC from './AllDataC';

Enzyme.configure({ adapter: new Adapter() })


describe("AllDataC", () => {

  it("should render my component", () => {
    const wrapper = shallow(<AllDataC data={[{ task: "Task-1", description: "Describe", status: "pending" }]} />);
    const tree = wrapper.debug();
    expect(tree).toMatchSnapshot();
  });

  it("button click test", () => {
    const pushMock = jest.fn();
    const wrapper = shallow(<AllDataC data={[{ task: "Task-1", description: "Describe", status: "pending" }]} history={{ push: pushMock }} />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'clickEdit');
    wrapper.find('#edit-btn').simulate('click');
    expect(instance.clickEdit).toHaveBeenCalledTimes(1);
  });

  it("button click test", () => {
    const pushMock = jest.fn();
    const wrapper = shallow(<AllDataC data={[{ task: "Task-1", description: "Describe", status: "pending" }]} history={{ push: pushMock }} />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'deleteClick');
    wrapper.find('#delete-btn').simulate('click');
    expect(instance.deleteClick).toHaveBeenCalledTimes(1);
  });
})