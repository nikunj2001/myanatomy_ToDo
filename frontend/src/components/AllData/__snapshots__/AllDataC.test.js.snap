// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllDataC should render my component 1`] = `
"<Fragment>
  <table border=\\"1\\" cellPadding=\\"5\\" cellSpacing=\\"5\\" width=\\"75%\\">
    <thead>
      <tr>
        <th>
          Task
        </th>
        <th>
          Description 
        </th>
        <th>
          Status
        </th>
        <th>
          Change Status
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Task-1
        </td>
        <td>
          Describe
        </td>
        <td>
          pending
        </td>
        <td>
          <button onClick={[Function: onClick]} className=\\"btn\\" id=\\"edit-btn\\">
            <BsFillPenFill className=\\"btns\\" />
          </button>
          <button className=\\"btn\\" onClick={[Function: onClick]} id=\\"delete-btn\\">
            <BsTrash className=\\"btns\\" />
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</Fragment>"
`;
